lib/parliament/ntriple/utils.rb

```diff
@@ -94,6 +95,27 @@ def self.sort_defaults
         { prepend_rejected: true }
       end
 
+      def self.multi_direction_sort(args)
+        directions_hash = { asc: 1, desc: -1 }
+        rejected = []
+        args = sort_defaults.merge(args)
+
+        list = args[:list].dup
+        sort_directions = args[:parameters]
+
+        list, rejected = prune_list(list, rejected, sort_directions.keys)
+        list.sort! do |obj1, obj2|
```

Comment:
We need to look at the efficiency of this algorithm - do we understand the potential complexity?